### PR TITLE
improve grammar/spelling in README

### PR DESCRIPTION
**Pull request by ScottRudiger**
Love this proposal and just wanted to help make the README a bit more clear--happy to update if there's disagreement on any of these changes

### CODE REVIEW

**Comment by ScottRudiger** [comment]
> -The following languages **implements** the operator with the same general semantics as this proposal (i.e., **guard** against a null base value, and short-circuiting **applying** to the whole chain):
> +The following languages **implement** the operator with the same general semantics as this proposal (i.e., **1) guarding** against a null base value, and **2)** short-circuiting **application** to the whole chain):

Just in case the reason for this change isn't clear, originally, it could've been read as:
`guard against both: 1) a null base value and 2) short-circuiting a whole chain`
Hopefully, it's now more clear that what is meant is:
`1) guard against a null base value and 2) allow short-circuiting a whole chain`

File: README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ Current Stage:
 * Gabriel Isenberg (@the_gisenberg)
 
 ## Overview and motivation
-When looking for a property value deeply in a tree structure, one has often to check whether intermediate nodes exist:
+When looking for a property value deep in a tree-like structure, one often has to check whether intermediate nodes exist:
```
Comment (ljharb) [inline README.md]: “deeply” is correct here; the other changes look good

File: README.md [hunk 2]
```diff
@@ -145,17 +145,17 @@ a == null ? undefined : delete a.b      // this is what we get, really
 
 ## Not supported
 
-The following are not supported for lack of real-world use cases:
+The following are not supported due to lack of real-world use cases:
```
Comment (ljharb) [inline README.md]: This is also correct as-is; this change needs an indefinite article a la “due to a lack”

File: README.md (hunk 1 above)
Comment (Kerrick) [reply README.md]: That depends -- are you "looking deeply" for a property value, or are you looking for a property value that is "deep" in the structure?

Comment (ljharb) [reply README.md]: Fair point; i guess I’d thought of the former but the latter is also true.

File: README.md (hunk 2 above)
Comment (ScottRudiger) [reply README.md]: You're right. "[D]ue to a lack of real-world use cases:" it is.

File: README.md (hunk 1 above)
Comment (ScottRudiger) [reply README.md]: Yeah, I'd agree "looking deeply" is possible, but "looking for a value that is deep in the structure" sounds more fitting in this situation, I think.

Would it help if instead it read, "When looking for a property value **that's** deep in a tree-like structure, one often has to check whether intermediate nodes exist:"?

Comment (ljharb) [reply README.md]: Sounds great!

**Pull request merged**